modules/react/button/lib/BaseButton.tsx

```diff
@@ -55,10 +55,7 @@ export interface ButtonContainerProps extends Partial<SystemIconProps>, GrowthBe
  */
 export interface BaseButtonProps extends Omit<ButtonContainerProps, 'ref'> {}
 
-/**
- * Temporary css variables to be used across all Buttons.
- */
-export const buttonVars = {
+export const buttonColorPropVars = {
```

Comment:
I think we still want a comment about these variables. The purpose of this object is for the `colors` prop - to provide backwards compatibility with how we allowed color overrides in Emotion. There is a better way now, directly using the variables directly and overriding what needs to be overridden in a style function call:

```ts
import {buttonStencil} from '@workday/canvas-kit-react/button';
import {systemIconStencil} from '@workday/canvas-kit-react/icon';
import {createStyles} from '@workday/canvas-kit-styling';

const myButtonStyles = createStyles({
  [buttonStencil.vars.background]: baseBackgroundColorOverride,
  [buttonStencil.vars.border]: baseBorderColorOverride,
  [buttonStencil.vars.boxShadowInner]: baseBoxShadowInnerColorOverride,
  [buttonStencil.vars.boxShadowOuter]: baseBoxShadowOuterColorOverride,
  [buttonStencil.vars.borderRadius]: baseBorderRadiusOverride,
  [buttonStencil.vars.label]: baseLabelColorOverride,
  [systemIconStencil.vars.color]: baseIconColorOverride,

  '&:focus-visible, &.focus': {
    [buttonStencil.vars.background]: focusBackgroundColorOverride,
    // etc
  },

  '&:hover, &.hover': {
    [buttonStencil.vars.background]: hoverBackgroundColorOverride,
    // etc
  },


  '&:active, &.active': {
    [buttonStencil.vars.background]: activeBackgroundColorOverride,
    // etc
  },


  '&:disabled, &.disabled': {
    [buttonStencil.vars.background]: disabledBackgroundColorOverride,
    // etc
  },

})
```

We could make a helper function that is structured outputs this that takes in a `colors` argument that is structured like the existing one that can be used to merge into style definitions if we think it's helpful.